### PR TITLE
Next release

### PR DESCRIPTION
**Pull request by Keats**

### CODE REVIEW

**Comment by xFrednet** [comment]
It might be good if we also add some documentation to the rust docs them self in this version. I would suggest that we simply add the example from the README and a link to it along with a list of available validators. 

My suggestion would be:
image

This is the current view:
image

I can create a PR for that documentation if you like :)

**Comment by Keats** [comment]
Please do!

**Comment by xFrednet** [comment]
Awesome, I'll do that when #132 is merged to avoid merge conflicts :)

File: validator_derive/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "validator_derive"
-version = "0.11.0"
+version = "0.12.0"
```
Comment (xFrednet) [inline validator_derive/Cargo.toml:3]: The version should also be bumped up in the `README.md`. I did that in #134

File: validator/src/validation/range.rs [hunk 2]
```diff
@@ -1,67 +1,59 @@
...
-        assert_eq!(validate_range(validator, 1_f64), true);
-    }
-
-    #[test]
-    fn test_validate_range_fail() {
-        let validator = Validator::Range { min: Some(0.0), max: Some(10.0) };
-        assert_eq!(validate_range(validator, 20_f64), false);
-    }
+    // This is needed to suppress the `validate_range` depreciation notice
+    // We want to keep the tests to make sure that this function will remain valid
```
Comment (xFrednet) [inline validator/src/validation/range.rs:27]: These comments are out of date they explained the `[allow(depeciated)]` from #132. These comments have also been removed in #134

File: README.md [hunk 3]
```diff
@@ -342,6 +342,12 @@ For example, the following attributes all work:
 
 ### validator
 
+## 0.12.0 (unreleased)
+
+- Allow `length` and `range` validators to take a reference to a variable
+- Make validator work with `Option<Vec<_>>`
+- Support reference for length types
```
Comment (xFrednet) [inline README.md:349]: Do we also want to add that #124 is fixed?
```md
- Fixed #124: The feature flags `phone`, `card`, `unic` can be used with derive again
```

**Review by xFrednet — COMMENTED** [review]
The rest looks good so far. #134 rounds everything up from my point of view :)

Thank you!

**Comment by Keats** [comment]
Thanks for the work @xFrednet !
I'm fixing some cyclical dependencies (https://github.com/Keats/validator/pull/135) and i'll publish it

**Pull request merged**